pkg/scheduler/objects/application.go

```diff
@@ -990,7 +990,7 @@ func (sa *Application) tryAllocate(headRoom *resources.Resource, preemptionDelay
 			// the iterator might not have the node we need as it could be reserved, or we have not added it yet
 			node := getNodeFn(requiredNode)
 			if node == nil {
-				log.Log(log.SchedApplication).Warn("required node is not found (could be transient)",
+				log.Log(log.SchedApplication).Debug("required node is not found (could be transient)",
```

Comment:
Could you use a rate-limit approach? It's still an issue if someone decides to use Debug level.

Reply:
> Could you use a rate-limit approach? It's still an issue if someone decides to use Debug level.
> 
> I propose a natural single msg/second limit.

I've added a RateLimitedLogger in the latest commit, appreciated for the comment.